src/swapProps.ts

```diff
@@ -0,0 +1,66 @@
+import { purry } from './purry';
+
+type SwappedProps<T, K1 extends keyof T, K2 extends keyof T> = {
+  [K in keyof T]: T[K1 extends K ? K2 : K2 extends K ? K1 : K];
+};
+
+/**
+ * @param data the object to be manipulated
+ * @param key1 the first property key
+ * @param key2 the second property key
+ *
+ * @signature
+ *   swap(key1, key2)(data)
+ *
+ * @returns Returns the manipulated object.
+ *
+ * @example
+ *   swap({a: 1, b: 2, c: 3}, 'a', 'b') // => {a: 2, b: 1, c: 3}
+ *
+ * @data_first
+ */
+export function swapProps<
+  T extends object,
+  K1 extends keyof T,
+  K2 extends keyof T
+>(data: T, key1: K1, key2: K2): SwappedProps<T, K1, K2>;
+
+/**
+ * Swaps the positions of two properties in an object based on the provided keys.
+ * @param key1 the first property key
+ * @param key2 the second property key
+ *
+ * @signature
+ *   swap(key1, key2)(data)
+ *
+ * @example
+ *   swap('a', 'b')({a: 1, b: 2, c: 3}) // => {a: 2, b: 1, c: 3}
+ *
+ * @returns Returns the manipulated object.
+ *
+ * @data_last
+ */
+export function swapProps<
+  T extends object,
+  K1 extends keyof T,
+  K2 extends keyof T
+>(key1: K1, key2: K2): (data: T) => SwappedProps<T, K1, K2>;
```

Comment:
This was an intentional change. data-last functions are primarily built for pipes, there's not a lot of value in having a standalone function factory that takes some params but leaves the actual object they work on generic. If we want to enforce that the prop names used are strictly coupled to the object they are manipulating (as with the data-first impl) then we have to configure the keys in relation to the main type. Using PropertyKey means they can take any value.

the TL;DR is that in order to get IDE auto-complete, and to get compile time protection against using the wrong key, we have to type the function like this. Otherwise this is valid and won't surface the error:
```ts
pipe(
  { a: "hello", b: "world" },
  swapProps("aTypo", "b")
);
```ts

Reply:
Thank you so much for all the excellent reviews!

Reply:
Thanks for staying patient and contributing to the project!